lua/nvim-tree/api.lua

```diff
@@ -0,0 +1,111 @@
+Api.tree.search_node = require("nvim-tree.actions.finders.search-node").fn
+Api.tree.collapse_all = require("nvim-tree.actions.tree-modifiers.collapse-all").fn
+Api.tree.expand_all = inject_node(require("nvim-tree.actions.tree-modifiers.expand-all").fn)
+Api.tree.toggle_gitignore_filter = require("nvim-tree.actions.tree-modifiers.toggles").git_ignored
+Api.tree.toggle_custom_filter = require("nvim-tree.actions.tree-modifiers.toggles").custom
+Api.tree.toggle_hidden_filter = require("nvim-tree.actions.tree-modifiers.toggles").dotfiles
+Api.tree.toggle_help = require("nvim-tree.actions.tree-modifiers.toggles").help
+
+Api.fs.create = inject_node(require("nvim-tree.actions.fs.create-file").fn)
+Api.fs.remove = inject_node(require("nvim-tree.actions.fs.remove-file").fn)
+Api.fs.trash = inject_node(require("nvim-tree.actions.fs.trash").fn)
+Api.fs.rename = inject_node(require("nvim-tree.actions.fs.rename-file").fn(false))
+Api.fs.rename_sub = inject_node(require("nvim-tree.actions.fs.rename-file").fn(true))
+Api.fs.copy = inject_node(require("nvim-tree.actions.fs.copy-paste").copy)
+Api.fs.cut = inject_node(require("nvim-tree.actions.fs.copy-paste").cut)
+Api.fs.paste = inject_node(require("nvim-tree.actions.fs.copy-paste").paste)
+Api.fs.print_clipboard = require("nvim-tree.actions.fs.copy-paste").print_clipboard
+Api.fs.copy.absolute_path = inject_node(require("nvim-tree.actions.fs.copy-paste").copy_absolute_path)
+Api.fs.copy.filename = inject_node(require("nvim-tree.actions.fs.copy-paste").copy_filename)
+Api.fs.copy.relative_path = inject_node(require("nvim-tree.actions.fs.copy-paste").copy_path)
+
+local function edit(mode, node)
+  local path = node.absolute_path
+  if node.link_to and not node.nodes then
+    path = node.link_to
+  end
+  require("nvim-tree.actions.node.open-file").fn(mode, path)
+end
+
+local function open_or_expand_or_dir_up(mode)
+  return function(node)
+    if node.name == ".." then
+      require("nvim-tree.actions.root.change-dir").fn ".."
+    elseif node.nodes then
+      require("nvim-tree.lib").expand_or_collapse(node)
+    else
+      edit(mode, node)
+    end
+  end
+end
+
+local function open_preview(node)
+  if node.nodes or node.name == ".." then
+    return
+  end
+
+  edit("preview", node)
+end
+
+Api.node.open.edit = inject_node(open_or_expand_or_dir_up "edit")
+Api.node.open.vertical = inject_node(open_or_expand_or_dir_up "vsplit")
+Api.node.open.horizontal = inject_node(open_or_expand_or_dir_up "split")
+Api.node.open.tab = inject_node(open_or_expand_or_dir_up "tabnew")
+Api.node.open.preview = inject_node(open_preview)
+
+Api.node.show_info_popup = inject_node(require("nvim-tree.actions.node.file-popup").toggle_file_info)
+Api.node.run.cmd = inject_node(require("nvim-tree.actions.node.run-command").run_file_command)
+Api.node.run.system = inject_node(require("nvim-tree.actions.node.system-open").fn)
+Api.node.navigate.sibling.next = inject_node(require("nvim-tree.actions.moves.sibling").fn "next")
```

Comment:
i agree it could be simpler, but since i don't expect all users to understand how closures work, it might get a bit complicated for them.
Also, this allows us to change the location and interfaces of the underlying code if such a change was made.

Reply:
Works for me.